src/shareon.js

```diff
@@ -27,10 +27,11 @@ const urlBuilderMap = {
   reddit: (d) => `https://www.reddit.com/submit?title=${d.title}&url=${d.url}`,
   teams: (d) => `https://teams.microsoft.com/share?href=${d.url}${d.text ? `&msgText=${d.text}` : ''}`,
   telegram: (d) => `https://telegram.me/share/url?url=${d.url}${d.text ? `&text=${d.text}` : ''}`,
+  tumblr: (d) => `https://www.tumblr.com/widgets/share/tool?posttype=${d.url}&tags=${d.hashtags}${d.title? `&title=${d.title}`:''}&content=${d.content}${d.text? `&caption=${d.text}`:''}${d.via? `&show-via=${d.via}`:''}`,
```

Comment:
This still requires changes:

- `posttype` should be literally `link`
- `d.title` always exists, so we do not need to check it
- `d.tags` may or may not exist, so we need to check for this (see how Twitter button does it)
- `d.content` does not exist and should be `d.url` instead

```suggestion
  tumblr: (d) => `https://www.tumblr.com/widgets/share/tool?posttype=link${d.hashtags? `&tags=${d.hashtags}` : ''}&title=${d.title}&content=${d.url}${d.text? `&caption=${d.text}`:''}${d.via? `&show-via=${d.via}`:''}`,
```